pkg/custompluginmonitor/plugin/plugin.go

```diff
@@ -234,7 +238,11 @@ func (p *Plugin) run(rule cpmtypes.CustomRule) (exitStatus cpmtypes.Status, outp
 	output = string(stdout)
 	output = strings.TrimSpace(output)
 
-	if cmd.ProcessState.Sys().(syscall.WaitStatus).Signaled() {
+	m.Lock()
+	wasKilled := killed
+	m.Unlock()
+
+	if cmd.ProcessState.Sys().(syscall.WaitStatus).Signaled() || wasKilled {
```

Comment:
Is `cmd.ProcessState.Sys().(syscall.WaitStatus).ExitStatus()` below supported on windows?

If the customer plugin is not supported on windows yet, should we avoid building the whole custom plugin part on windows for now?

Reply:
Just build with `disable_custom_plugin_monitor`.

Reply:
I'd like to keep it for now since it can work. Also this logic works fine.
The ExitStatus does not work. It's pinned to false. Once you get into syscall territory you will have to confirm each OS. 

There's a potential use for this logic see the design document for the use cases.

Reply:
If this is the case, maybe just always use the `timeout` variable? That makes the logic more OS independent.